src/Sfx/App/Scripts/Models/DataModels/Collections.ts

```diff
@@ -61,27 +115,54 @@ module Sfx {
             return "uniqueId";
         }
 
-        public constructor(public data: DataService, parent?: any) {
+        public constructor(public data: DataService, parent?: any, appendOnly: boolean = false) {
             this.parent = parent;
+            this.appendOnly = appendOnly;
+            this.refreshingLoadPromise = new CancelablePromise(this.data.$q);
         }
 
         // Base refresh logic, do not override
         public refresh(messageHandler?: IResponseMessageHandler): angular.IPromise<any> {
             if (!this.refreshingPromise) {
-                this.refreshingPromise = this.retrieveNewCollection(messageHandler).then(collection => {
-                    return this.update(collection);
-                }).then(() => {
-                    return this;
-                }).finally(() => {
-                    this.refreshingPromise = null;
-                });
+                this.refreshingPromise =
+                    this.refreshingLoadPromise.load(() => {
+                        return this.retrieveNewCollection(messageHandler);
+                    }).catch((error) => {
+                        if (!error || error.isCanceled !== true) {
+                            throw error;

```

Comment:
Not sure if this is by intention - throw error when !error is true?

Reply:
I just wanted to handle here the case where a different error other than cancellation happens. I believe in this case the passed object will always have a value, so I'll be changing this to error && error.isCanceled !== true.